sdk/servicebus/service-bus/test/atomManagement.spec.ts

```diff
@@ -47,364 +48,444 @@ const alwaysBeDeletedSubscription = "alwaysbedeletedsubscription";
-            throw new Error("TestError: Unrecognized EntityType");
-        }
+      try {
+        await deleteEntity(
+          testCase.entityType,
+          testCase.alwaysBeDeletedEntity,
+          testCase.parentTopicName,
+          testCase.parentSubscriptionName
+        );
+      } catch (err) {
+        log.httpAtomXml("Ignoring clean up step");
+      }
 
-        if (!skipTest) {
-          should.equal(
-            response.createdAt,
-            undefined,
-            "Should receive empty result and just the raw response"
-          );
-        }
-        // Error is undefined for queues, topics - but not for non-existent subscriptions and rules
-      });
+      should.equal(Array.isArray(response), true, "Result must be any array for list requests");
+      should.equal(
+        response.length,
+        testCase.entityType == EntityType.RULE ? 2 : 1,
+        "Result must be an empty array"
+      );
+      should.equal(
+        response[0][testCase.entityType.toLowerCase() + "Name"],
+        testCase.entityType == EntityType.RULE
+          ? testCase.alwaysBeDeletedEntity
+          : testCase.alwaysBeExistingEntity,
+        "Entity name mismatch"
+      );
+    });
+
+    it(`Creates a non-existent ${testCase.entityType} entity successfully`, async () => {
+      let response;
+
+      try {
+        response = await deleteEntity(
+          testCase.entityType,
+          testCase.alwaysBeExistingEntity,
+          testCase.parentTopicName,
+          testCase.parentSubscriptionName
+        );
+      } catch (err) {
+        log.httpAtomXml("Ignoring clean up step");
+      }
+      response = await createEntity(
+        testCase.entityType,
+        testCase.alwaysBeExistingEntity,
+        testCase.parentTopicName,
+        testCase.parentSubscriptionName
+      );
+      should.equal(
+        response[testCase.entityType.toLowerCase() + "Name"],
+        testCase.alwaysBeExistingEntity,
+        "Entity name mismatch"
+      );
+    });
+
+    it(`Creating an existent ${testCase.entityType} entity throws an error`, async () => {
+      let response;
+
+      try {
+        await createEntity(
+          testCase.entityType,
+          testCase.alwaysBeExistingEntity,
+          testCase.parentTopicName,
+          testCase.parentSubscriptionName
+        );
+      } catch (err) {
+        response = err;
+      }
+
+      should.equal(response.statusCode, 409, "Error must not be undefined");
+      should.equal(
+        response.message.startsWith("The messaging entity") ||
+          response.message.startsWith("Entity") ||
+          response.message.startsWith("SubCode") ||
+          response.message.startsWith("No service"),
+        true,
+        `Expected error message to be a textual content but got "${response.message}"`
+      );
     });
-  }
-);
+
+    it(`Lists available ${testCase.entityType} entities successfully`, async () => {
+      let response;
+
+      response = await listEntities(
```

Comment:
I think we should create some entities first before listing to be sure there are entities to list. If I ran this test in isolation (it.only) it should work.

Then we can also test that the array length is > 0. (We already have a test that ensures we can get an empty list, would be good to verify we can get a non-empty list which I think is what this test is for).

Reply:
Updated